src/metabase/email/messages.clj

```diff
@@ -393,24 +411,42 @@
 (defn- result-attachments [results]
   (filter some? (mapcat result-attachment results)))
 
-(defn- render-message-body [message-template message-context timezone results]
-  (let [rendered-cards (binding [render/*include-title* true]
-                         (mapv #(render/render-pulse-section timezone %) results))
-        message-body   (assoc message-context :pulse (html (vec (cons :div (map :content rendered-cards)))))
-        attachments    (apply merge (map :attachments rendered-cards))]
-    (vec (concat [{:type "text/html; charset=utf-8" :content (stencil/render-file message-template message-body)}]
-                 (map make-message-attachment attachments)
-                 (result-attachments results)))))
+(defn- render-result-card
+  [timezone result]
+  (if (:card result)
+    (render/render-pulse-section timezone result)
+    {:content (markdown/process-markdown (:text result) :html)}))
+
+(defn- render-message-body [message-type message-context timezone dashboard results]
+  (let [rendered-cards  (binding [render/*include-title* true]
+                          (mapv #(render-result-card timezone %) results))
+        icon-name       (case message-type
+                          :alert :bell
+                          :pulse :dashboard)
+        icon-attachment (first (map make-message-attachment (icon-bundle icon-name)))
+        message-body    (assoc message-context :pulse   (html (vec (cons :div (map :content rendered-cards))))
+                                               :iconCid (:content-id icon-attachment))
+        attachments     (apply merge (map :attachments rendered-cards))]
```

Comment:
apply merge over attachments seems wrong to me here. Wouldn't you want to concat them, not merge them?

Reply:
Oh i see now. they are maps of `{id image-url}` so merging is indeed correct.

Reply:
yep, this line is actually unmodified from how it was before